newton/_src/utils/selection.py

```
@@ -153,58 +231,230 @@ def __init__(
+        counts_per_world = [0] * world_count
+        for world_id in range(world_count):
+            count = len(articulation_ids[world_id])
+            counts_per_world[world_id] += count
+            articulation_count += count
 
-        arti_joint_begin = model_articulation_start[arti_0]
-        arti_joint_end = model_articulation_start[arti_0 + 1]  # FIXME: is this always correct?
-        arti_joint_count = arti_joint_end - arti_joint_begin
-        arti_link_count = arti_joint_count
+        if articulation_count == 0:
+            raise KeyError("No matching articulations")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)